src/nvim/mbyte.c

```diff
@@ -3406,22 +3406,23 @@ static int enc_alias_search(char_u *name)
  */
 char_u * enc_locale(void)
 {
-  char        *s;
-  char        *p;
   int i;
   char buf[50];
+
+  const char *s;
 # ifdef HAVE_NL_LANGINFO_CODESET
-  if ((s = nl_langinfo(CODESET)) == NULL || *s == NUL)
+  if (strempty(s = nl_langinfo(CODESET)))
```

Comment:
It makes sense to me as well, but I still think it would be clearer as the assignments below don't involve function calls inside arguments in addition to the if. Up to you.